src/renderer/shared/ui-kit/Dropdown/Dropdown.tsx

```diff
@@ -38,27 +40,47 @@ const Root = ({
   testId = 'Dropdown',
   children,
 }: RootProps) => {
+  const [isOpen, toggleIsOpen] = useToggle(open);
+
   const ctx = useMemo(
-    () => ({ preventClosing, side, sideOffset, align, alignOffset, testId }),
-    [preventClosing, side, sideOffset, align, alignOffset, testId],
+    () => ({ isOpen, preventClosing, side, sideOffset, align, alignOffset, testId }),
+    [isOpen, preventClosing, side, sideOffset, align, alignOffset, testId],
+  );
+
+  const openChange = useCallback(
+    (isOpen: boolean) => {
+      onToggle?.(isOpen);
+      toggleIsOpen();
+    },
+    [onToggle],
   );
 
   return (
     <Context.Provider value={ctx}>
-      <DropdownMenu.Root open={open} modal onOpenChange={onToggle}>
+      <DropdownMenu.Root modal open={open} onOpenChange={openChange}>
         {children}
       </DropdownMenu.Root>
     </Context.Provider>
   );
 };
 
-const Trigger = ({ children }: PropsWithChildren) => <DropdownMenu.Trigger asChild>{children}</DropdownMenu.Trigger>;
+const Trigger = ({ children }: { children: ((open: boolean) => ReactNode) | ReactNode }) => {
+  const { isOpen } = useContext(Context);
+
+  return (
+    <DropdownMenu.Trigger asChild>
+      {typeof children === 'function' ? children(Boolean(isOpen)) : children}
```

Comment:
In which cases do we need logic like this?

Reply:
Component can pass this value to children prop, so they can observe whether dropdown is open or not
In my case it's useful for chevron icon rotation